### PR TITLE
Add text block edit functionality

### PR DESCRIPTION
**Pull request by MrNossiom**
Hi 👋🏻,

To add the ability of editing text blocks, I quickly duplicated the `EditBookmarkCard` file and modified the `.../{digestId}/block/{blockId}` endpoint `PATCH` implementation. Adding an `onEditClick` prop to the `BlockTextCard` component to show the `Edit` button in the Block Menu Popover.

Didn't test intensively, but it should work fine 😄 

## Improvements
- Merge `AddTextBlockDialog` and `EditTextBlockDialog` into one component?
- Cleanup the `.../{digestId}/block/{blockId}` file to avoid nesting. Factorize `if-else` statement in functions?

### CODE REVIEW

**Comment by shinework** [comment]
Feature needed! 🙏  Thanks @MrNossiom

**Comment by shinework** [comment]
Hello @MrNossiom! We have slightly updated the UI related to your PR, see https://github.com/premieroctet/digestclub/pull/15

**Comment by MrNossiom** [comment]
I changed the button label of the `AddTextBlockDialog` from `Save` to `Add`. Just as a suggestion, I can revert it in seconds 😉.

File: src/components/digests/dialog/EditTextBlockDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  text: string;
+}
+interface Props {
+  isOpen: boolean;
+  setIsOpen: (isOpen: boolean) => void;
+  bookmarkDigest: BookmarkCardProps['block'];
+  defaultValues?: Partial<FormValues>;
+}
+
+export default function EditBookmarkDialog({
```
Comment (shinework) [inline src/components/digests/dialog/EditTextBlockDialog.tsx:27]:
```suggestion
export default function EditTextBlockDialog({
```

File: src/components/digests/dialog/EditTextBlockDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    'update-bookmarkdigest',
+    ({ text }) => {
+      return api.patch(
+        `/teams/${teamId}/digests/${params?.digestId}/block/${bookmarkDigest.id}`,
+        { text }
+      );
+    },
+    {
+      onSuccess: () => {
+        successToast('Bookmark updated');
```
Comment (shinework) [inline src/components/digests/dialog/EditTextBlockDialog.tsx:61]:
```suggestion
        successToast('Digest updated');
```

**Review by shinework — CHANGES_REQUESTED** [review]

**Comment by shinework** [comment]
Thanks @MrNossiom! I have spot a bug, when the user edits the text and re-edit, the textarea still contains the old text : 

https://github.com/premieroctet/digestclub/assets/1102595/e0088c79-b76c-4860-ac91-3c376073e09c

**Comment by MrNossiom** [comment]
> I have spot a bug, when the user edits the text and re-edit, the textarea still contains the old text

Good catch 👍🏻

**Comment by shinework** [comment]
Thanks a lot @MrNossiom! :shipit:

**Review by shinework — APPROVED** [review]

**Pull request merged**